MediaTracker/app/src/main/java/com/izor066/android/mediatracker/ui/adapter/ItemAdapter.java

```diff
@@ -0,0 +1,90 @@
+package com.izor066.android.mediatracker.ui.adapter;
+
+
+import android.content.Context;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.izor066.android.mediatracker.MediaTrackerApplication;
+import com.izor066.android.mediatracker.R;
+import com.izor066.android.mediatracker.api.DataSource;
+import com.izor066.android.mediatracker.api.model.Book;
+import com.squareup.picasso.Picasso;
+
+/**
+ * Created by igor on 13/11/15.
+ */
+public class ItemAdapter extends RecyclerView.Adapter<ItemAdapter.ItemAdapterViewHolder>  {
+
+    public interface OnBookClickListener {
+         void onBookClick(Book book);
+    }
+    
+    private final OnBookClickListener listener;
+    
+    public ItemAdapter(OnBookClickListener listener) {
+        this.listener = listener;
+    }
+
+    @Override
+    public ItemAdapterViewHolder onCreateViewHolder(ViewGroup viewGroup, int index) {
+        View inflate = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.book, viewGroup, false);
+        return new ItemAdapterViewHolder(inflate);
+    }
+
+    @Override
+    public void onBindViewHolder(ItemAdapterViewHolder itemAdapterViewHolder, int i) {
+        DataSource sharedDataSource = MediaTrackerApplication.getSharedDataSource();
+        itemAdapterViewHolder.update(sharedDataSource.getAllBooks().get(i), listener);
+    }
+
+    @Override
+    public int getItemCount() {
+        return MediaTrackerApplication.getSharedDataSource().getAllBooks().size();
+    }
+
+    class ItemAdapterViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
```

Comment:
This inner class should be `static`. It is advisable inner classes to be `static` so that they don't keep a reference to the outer class thus preventing the Garbage Collector to be able to collect the outer class when not needed.